databricks/koalas/frame.py

```diff
@@ -3497,6 +3497,132 @@ def append(self, other: 'DataFrame', ignore_index: bool = False,
+
+        See Also
+        --------
+        DataFrame.merge : For column(s)-on-columns(s) operations.
+
+        Examples
+        --------
+        >>> left_kdf = ks.DataFrame({'A': ['1', '2', '3'], 'B': ['100', '200', '300']},
+        ...                         columns=['A', 'B'])
+        >>> right_kdf = ks.DataFrame({'B': ['x', 'y', 'z'],'C': ['d', 'e', 'f']},
+        ...                          columns=['B', 'C'])
+        >>> left_kdf
+           A    B
+        0  1  100
+        1  2  200
+        2  3  300
+        >>> right_kdf
+           B  C
+        0  x  d
+        1  y  e
+        2  z  f
+        >>> left_kdf.update(right_kdf)
+        >>> left_kdf
+           A  B
+        0  1  x
+        1  2  y
+        2  3  z
+
+        The DataFrame's length does not increase as a result of the update,
+        only values at matching index/column labels are updated.
+
+        >>> left_kdf = ks.DataFrame({'A': ['1', '2', '3'], 'B': ['100', '200', '300']},
+        ...                         columns=['A', 'B'])
+        >>> right_kdf = ks.DataFrame({'B': ['u', 'v', 'w', 'x', 'y', 'z']}, columns=['B'])
+        ...
+        >>> left_kdf.update(right_kdf)
+        >>> left_kdf
+           A  B
+        0  1  u
+        1  2  v
+        2  3  w
+
+        For Series, it's name attribute must be set.
+
+        >>> left_kdf = ks.DataFrame({'A': ['1', '2', '3'], 'B': ['100', '200', '300']},
+        ...                         columns=['A', 'B'])
+        >>> right_kser = ks.Series(['x', 'z'], name='B', index=[0, 2])
+        >>> left_kdf.update(right_kser)
+        >>> left_kdf
+           A    B
+        0  1    x
+        1  2  200
+        2  3    z
+
+        If `right` contains NaNs the corresponding values are not updated
+        in the original dataframe.
+
+        >>> left_kdf = ks.DataFrame({'A': ['1', '2', '3'], 'B': ['100', '200', '300']}
+        ...                         , columns=['A', 'B'])
+        >>> right_kdf = ks.DataFrame({'B': ['x', None, 'z']}, columns=['B'])
+        >>> left_kdf.update(right_kdf)
+        >>> left_kdf
+           A    B
+        0  1    x
+        1  2  200
+        2  3    z
+        """
+        if errors not in ['ignore', 'raise']:
+            raise ValueError("The parameter errors must be either 'ignore' or 'raise'")
+
+        if join != 'left':
+            raise NotImplementedError("Only left join is supported")
+
+        if isinstance(right, ks.Series):
+            right = DataFrame(right)
+
+        update_columns = list(set(self._metadata.data_columns) & set(right._metadata.data_columns))
+        update_sdf = self.join(right[update_columns], rsuffix='_new')._sdf
+
+        for column_name in update_columns:
+            old_col = update_sdf[column_name]
+            new_col = update_sdf[column_name + '_new']
+            if errors == 'raise':
+                null_sdf = update_sdf.filter(old_col.isNull() & new_col.isNull())
```

Comment:
Seems like this will always throw an exception because   `update_sdf.filter(...)` always return a dataframe and never return `None`.